Assets/Scripts/Characters/EntityController.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EntityController : MonoBehaviour {
+
+    public float health { get; set; }
+    public float totalHealth { get; set; }
+
+    public bool immobile { get; set; }
+    public bool fixedFacing { get; set; }
+    public bool silent { get; set; }
+    public bool cantAttack { get; set; }
+
+    public float endStun;
+    public float endFreeze;
+
+    public IDictionary<Stat, float> baseStats;
+    public List<Effect> currentEffects;
+    public IDictionary<Stat, List<Modifier>> activeModifiers;
+    public IDictionary<Stat, float> effectiveStats;
+
+    protected Rigidbody charRigidbody;
+    protected Vector3 movement;
+
+    // Use this for initialization
+    protected virtual void Start () {
+        this.charRigidbody = GetComponent<Rigidbody> ();
+        this.immobile = false;
+        this.fixedFacing = false;
+        this.silent = false;
+        this.cantAttack = false;
+        this.currentEffects = new List<Effect> ();
+        this.baseStats = new Dictionary<Stat, float> ();
+        this.effectiveStats = new Dictionary<Stat, float> ();
+        this.activeModifiers = new Dictionary<Stat, List<Modifier>> ();
+        foreach (Stat s in System.Enum.GetValues(typeof(Stat))) {
```

Comment:
Might be nice to isolate this behavior in a new class, so rather than needing to maintain a dictionary of stats (which always will be pre-populated with 0 for all stats) we might have something like:

```
class EntityStats {

  private Dictionary<Stat, float> baseStats;
  private Dictionary<Stat, float> effectiveStats;
  private Dictionary<Stat, List<Modifier>> statModifiers;
  
  public EntityStats() {
    (initialize)
  }

  public void SetBaseStats(Dictionary<Stat, float> stats) {
    ...
  }

  public void ApplyStatModifier(Stat stat, Modifier modifier) {
    ...
  }
}
```